### ISSUE TITLE
architecture

### ISSUE THREAD

**Issue by durka** [SDBlumire/rust-lab#3]
There should probably be a folder in the TODO section for architecture. Both frontend and backend are in question. Are we going to use one of the Rust web frameworks? The one I know is iron. We probably want something in front, like nginx.

**Comment by tilpner** [comment]
NGinx would be specific to deployment, not rust-lab itself. Yes, it's needed, but that's the responsibility of anyone who deploys rust-lab.

Given the *Do as much of it as we can in rust* priority, an Iron backend seems like a good choice. The backend probably won't expose much, so any immaturity of the Rust web ecosystem isn't as bad, and even Iron could be replaced easily if there's just a few routes that call into library-independent code.

**Comment by SDBlumire** [comment]
I agree with using an Iron backend. As for frontend, probably just throwing together something in React or Angular to interface with an API exposed by our backend.
